producer/buffer/options.go

```diff
@@ -24,23 +24,27 @@ import (
 	"time"
 
 	"github.com/m3db/m3x/instrument"
+	"github.com/m3db/m3x/retry"
 )
 
 var (
-	defaultMaxBufferSize      = 100 * 1024 * 1024 // 100MB.
-	defaultMaxMessageSize     = 1 * 1024 * 1024   // 1MB.
-	defaultCleanupInterval    = 5 * time.Second
-	defaultCloseCheckInterval = time.Second
-	defaultScanBatchSize      = 256
+	defaultMaxBufferSize       = 100 * 1024 * 1024 // 100MB.
+	defaultMaxMessageSize      = 1 * 1024 * 1024   // 1MB.
+	defaultCloseCheckInterval  = time.Second
+	defaultScanBatchSize       = 16
+	defaultCleanupRetryOptions = retry.NewOptions().
```

Comment:
nit: might be cleaner to only set the initial backoff and maxbackoff here and construct the retry options in the ctor. These all look like constants rather than vars.

Reply:
I'll move the initial backoff and max backoff to const, I think it's more flexible to set a retry options here as user can config max retry times if they want, the forever can be set to false, etc.

Reply:
Hm not sure why this is less flexible if the logic is moved to the ctor but okay.

Reply:
I see what you mean now, I thought you meant to just set initialBackoff and maxBackoff in the options, yeah can move this to the constructor.